tools/cwrap/plugins/Broadcast.py

```diff
@@ -0,0 +1,294 @@
+from . import CWrapPlugin
+from string import Template
+
+# Arguments to the Broadcast Plugin:
+# broadcast: args_to_broadcast_against [inplace] [fallback]
+# [args_to_broadcast_against]: either a single argument (e.g. "arg1") or a comma-seperated
+#                              list of two arguments (e.g. "tensor1,tensor2") indicating
+#                              arguments to broadcast specified argument (usually "self") against
+# [inplace] will generate code for in-place function, which doesn't allow the in-place
+#           argument to be broadcast
+# [fallback] if tensors aren't broadcastable, preserves "element number" pointwise behavior,
```

Comment:
Nit: make this `broadcast_fallback` or something that makes it explicit what behavior we are "falling back" from.